test/Altinn.AccessManagement.Tests/Controllers/V2RightsInternalControllerTest.cs

```diff
@@ -0,0 +1,156 @@
+public class V2RightsInternalControllerTest(WebApplicationFixture fixture) : ControllerTest(fixture)
+{
+    private static Action<AcceptanceCriteriaTest> WithAssertDbLastDelegationToUserIsRevoked(IParty from, IUserProfile to) => test =>
+    {
+        test.ApiAssertions.Add(async fixture =>
+        {
+            var delegations = await fixture.Postgres.ListDelegationsChanges(filter => filter.Where(delegation => from.Party.PartyId == delegation.OfferedByPartyId && to.UserProfile.UserId == delegation.CoveredByUserId));
+
+            Assert.True(
+                delegations.OrderBy(delegation => delegation.Created).First().DelegationChangeType == DelegationChangeType.RevokeLast,
+                $"Last delegation from party '{from.Party.Name}' with party ID '{from.Party.PartyId}' to user '{to.UserProfile.Party.Name}' with user ID '{to.UserProfile.UserId}' is not of type '{DelegationChangeType.RevokeLast}'");
+        });
+    };
+
+    private static void WithAssertDbDelegationsNotEmpty(AcceptanceCriteriaTest test)
+    {
+        test.ApiAssertions.Add(async api =>
+        {
+            var delegations = await api.Postgres.ListDelegationsChanges();
+            Assert.NotEmpty(delegations);
+        });
+    }
+
+    private static void WithAssertEmptyDelegationList(AcceptanceCriteriaTest test)
+    {
+        test.ResponseAssertions.Add(async response =>
+        {
+            var delegations = await response.Content.ReadFromJsonAsync<IEnumerable<RightDelegationExternal>>();
+            Assert.Empty(delegations);
+        });
+    }
+
+    private static Action<AcceptanceCriteriaTest> WithAssertResponseContainsDelegationToUserProfile(IParty from, IUserProfile to) => test =>
+    {
+        test.ResponseAssertions.Add(async response =>
+        {
+            var delegations = await response.Content.ReadFromJsonAsync<IEnumerable<RightDelegationExternal>>();
+            var result = delegations.Any(delegation =>
+                    delegation.To.Any(attribute => attribute.Value == from.Party.PartyId.ToString()) &&
+                    delegation.From.Any(attribute => attribute.Value == to.UserProfile.UserId.ToString()));
+
+            Assert.True(result, $"Response don't contain any delegations from '{from.Party.Name}' with party ID '{from.Party.PartyId}' to user profile '{to.UserProfile.Party.Name}' with user ID '{to.UserProfile.UserId}'");
+        });
+    };
+
+    private static Action<AcceptanceCriteriaTest> WithAssertResponseContainsDelegationToParty(IParty from, IParty to) => test =>
+    {
+        test.ResponseAssertions.Add(async response =>
+        {
+            var delegations = await response.Content.ReadFromJsonAsync<IEnumerable<RightDelegationExternal>>();
+            var result = delegations.Any(delegation =>
+                delegation.To.Any(attribute => attribute.Value == from.Party.PartyId.ToString()) &&
+                delegation.From.Any(attribute => attribute.Value == to.Party.PartyId.ToString()));
+
```

Comment:
From/To mixup? 
delegation.To ... from.Party
delegation.From ... to.Party

Reply:
done